test/buffers_utest.cpp

```diff
@@ -63,222 +55,156 @@ class BuffersUtest {
         }
-        assert(ok);
     }
 
-    void buffers_utest3()
-    {
-        std::cout << "buffers_utest3\n";
-
-        int n = 8;
-        int size = 32;
-        int n1 = 4;
-        int n2 = 10;
-
-        quadiron::vec::Buffers<T>* vec = gen_buffers_rand_data(n, size);
-        quadiron::vec::Buffers<T> vec1(vec, n1);
-        quadiron::vec::Buffers<T> vec2(vec, n2);
-
-        quadiron::vec::Buffers<T> _vec1(vec, 0, n1);
-        quadiron::vec::Buffers<T>* _vec2 =
-            new quadiron::vec::BuffersZeroExtended<T>(vec, n2);
-
-        assert(vec1.eq(&_vec1));
-        assert(vec2.eq(_vec2));
-
-        delete vec;
-
-        quadiron::vec::Buffers<T> vec3(&vec2, n1);
-        assert(vec3.eq(&vec1));
-
-        delete _vec2;
+    std::vector<TypeParam*> mem3(end - begin, nullptr);
+    for (int i = 0; i < end - begin; i++) {
+        mem3[i] = mem1->at(i + begin);
     }
+    vec::Buffers<TypeParam> vec3(end - begin, size, &mem3);
 
-    void buffers_utest4()
-    {
-        std::cout << "buffers_utest4\n";
-        int i;
-        int word_size;
+    ASSERT_EQ(vec2, vec3);
+}
 
-        for (i = 0; i <= quadiron::arith::log2<T>(sizeof(T)); i++) {
-            word_size = quadiron::arith::exp2<T>(i);
-            pack_unpack(word_size);
+TYPED_TEST(BuffersTest, Test2) // NOLINT
+{
+    const int n = 8;
+    const int size = 32;
+    const int half = n / 2;
+    auto vec1 = this->gen_buffers_rand_data(n, size);
+    vec::Buffers<TypeParam> vec2(n, size);
+
+    vec2.copy(vec1.get());
+
+    std::vector<TypeParam*>* even_mem =
+        new std::vector<TypeParam*>(half, nullptr);
+    std::vector<TypeParam*>* odd_mem =
+        new std::vector<TypeParam*>(half, nullptr);
+    vec::Buffers<TypeParam> i_even(half, size, even_mem);
+    vec::Buffers<TypeParam> i_odd(half, size, odd_mem);
+    vec1->separate_even_odd(&i_even, &i_odd);
+
+    vec1->separate_even_odd();
+
+    vec::Buffers<TypeParam> _i_even(vec1.get(), 0, half);
+    vec::Buffers<TypeParam> _i_odd(vec1.get(), half, n);
+    ASSERT_EQ(i_even, _i_even);
+    ASSERT_EQ(i_odd, _i_odd);
+
+    const std::vector<TypeParam*>* mem1 = vec1->get_mem();
+    const std::vector<TypeParam*>* mem2 = vec2.get_mem();
+
+    bool ok = true;
+    for (int i = 0; i < n / 2; i += 2) {
+        TypeParam* even1 = mem1->at(i);
+        TypeParam* even2 = mem2->at(i * 2);
+        TypeParam* odd1 = mem1->at(i + n / 2);
+        TypeParam* odd2 = mem2->at(i * 2 + 1);
+        for (int j = 0; j < size; j++) {
+            if (even1[j] != even2[j] || odd1[j] != odd2[j]) {
+                ok = false;
+                i = n;
+                break;
+            }
         }
     }
+    ASSERT_TRUE(ok);
+}
 
-    void pack_unpack(int word_size)
-    {
-        std::cout << "pack_unpack with word_size=" << word_size << "\n";
-        int n = 8;
-        int size = 32;
-        int bytes_size = size * word_size;
-        int i, j, t, u;
+TYPED_TEST(BuffersTest, Test3) // NOLINT
```

Comment:
@lamphamsy If you have a better name than Test3, I'll be glad to rename it (it was named utest3 before so I kept it like that)

Reply:
Here is:  `Zero-extended constructor checking`

Reply:
Done